src/login/logind-dbus.c

```diff
@@ -2324,20 +2324,49 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
 
 static int method_cancel_scheduled_shutdown(sd_bus_message *message, void *userdata, sd_bus_error *error) {
         Manager *m = userdata;
+        const char *action;
         bool cancelled;
+        int r;
 
         assert(m);
         assert(message);
 
         cancelled = m->scheduled_shutdown_type != NULL;
```

Comment:
Could you drop `! = NULL`? Our coding style allows implicit cast from pointer to boolean.

Reply:
Makes sense for conditions, I overlooked that. Even in the assignment case though? Should I add a cast instead so it's clear we don't try to assign a pointer?